test/block/block.js

```diff
@@ -15,245 +15,237 @@ var dataJson = fs.readFileSync('test/data/blk642579-testnet.json').toString();
+     });
+
+     it('doesn\'t create a bufferWriter if one provided', function() {
+       var writer = new BufferWriter();
+       var block = Block.fromBuffer(blockbuf);
+       block.toBufferWriter(writer).should.equal(writer);
+     });
+
+   });
+
+   describe('#toObject', function() {
+
+     it('should recover a block from genesis block buffer', function() {
+       var block = Block.fromBuffer(blockOneBuf);
+       block.id.should.equal(blockOneId);
+       block.toObject().should.deep.equal({
+         header: {
+           hash: '00000000839a8e6886ab5951d76f411475428afc90947ee320161bbf18eb6048',
+           version: 1,
+           prevHash: '000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f',
+           merkleRoot: '0e3e2357e806b6cdb1f70b54c3a3a17b6714ee1f0e68bebb44a74b1efd512098',
+           time: 1231469665,
+           bits: 486604799,
+           nonce: 2573394689
+         },
+         transactions: [{
+           hash: '0e3e2357e806b6cdb1f70b54c3a3a17b6714ee1f0e68bebb44a74b1efd512098',
+           version: 1,
+           inputs: [{
+             prevTxId: '0000000000000000000000000000000000000000000000000000000000000000',
+             outputIndex: 4294967295,
+             sequenceNumber: 4294967295,
+             script: '04ffff001d0104'
+           }],
+           outputs: [{
+             satoshis: 5000000000,
+             script: '410496b538e853519c726a2c91e61ec11600ae1390813a627c66fb8be7947be63c' +
+               '52da7589379515d4e0a604f8141781e62294721166bf621e73a82cbf2342c858eeac'
+           }],
+           nLockTime: 0
+         }]
+       });
+     });
+
+     it('roundtrips correctly', function() {
+       var block = Block.fromBuffer(blockOneBuf);
+       var obj = block.toObject();
+       var block2 = Block.fromObject(obj);
+       block2.toObject().should.deep.equal(block.toObject());
+     });
+
+   });
+
+   describe('#_getHash', function() {
+
+     it('should return the correct hash of the genesis block', function() {
+       var block = Block.fromBuffer(genesisbuf);
+       var blockhash = new Buffer(Array.apply([], new Buffer(genesisidhex, 'hex')).reverse());
+       block._getHash().toString('hex').should.equal(blockhash.toString('hex'));
+     });
+   });
+
+   describe('#id', function() {
+
+     it('should return the correct id of the genesis block', function() {
+       var block = Block.fromBuffer(genesisbuf);
+       block.id.should.equal(genesisidhex);
+     });
+     it('"hash" should be the same as "id"', function() {
+       var block = Block.fromBuffer(genesisbuf);
+       block.id.should.equal(block.hash);
+     });
+
+   });
+
+
+   describe('#inspect', function() {
+     it('should return the correct inspect of the genesis block', function() {
+       var block = Block.fromBuffer(genesisbuf);
+       block.inspect().should.equal('<Block ' + genesisidhex + '>');
+     });
+   });
+   */
+
+  describe('#merkleRoot', function () {
```

Comment:
Try to test with the hex of a block with version = 3, I expect it to fails.

Reply:
Remove this test along with the method